### ISSUE TITLE
The difference between the global and local methods is not clear.

### ISSUE THREAD

**Issue by loySoGxj** [stellargraph/stellargraph#1982]
Is there a treatise that shows the difference between global and local search methods in edgeSplitter.train_test_split of stellargraph?

`edgeSplitter = EdgeSplitter(G)
graph_test, samples_test, labels_test = edgeSplitter.train_test_split(p=0.1, method="global", seed=24)`

**Comment by huonw** [comment]
The source code is unfortunately the best reference for this. The relevant file is https://github.com/stellargraph/stellargraph/blob/e46af15c2dbcaa9ae6328dd1ec62c6f9ef7f5467/stellargraph/data/edge_splitter.py, and searching for `local` or `global` reveals:

https://github.com/stellargraph/stellargraph/blob/e46af15c2dbcaa9ae6328dd1ec62c6f9ef7f5467/stellargraph/data/edge_splitter.py#L125-L141

The two methods called there have doc strings:

Global:

https://github.com/stellargraph/stellargraph/blob/e46af15c2dbcaa9ae6328dd1ec62c6f9ef7f5467/stellargraph/data/edge_splitter.py#L914-L923

Local:

https://github.com/stellargraph/stellargraph/blob/e46af15c2dbcaa9ae6328dd1ec62c6f9ef7f5467/stellargraph/data/edge_splitter.py#L804-L819

Example: consider a really simple line graph with 8 nodes `a -- b -- c -- d -- e -- f -- g -- h`. Suppose the sampler chooses `a` as the starting node for a negative edge:

- global: any other node (`c`, `d`, ... `h`) can be chosen for the second node to make the edge, meaning it can generate a negative edge like `a -- f`
- local: only nodes close to the start `a` can be chosen. With the default probabilities, it's up to 3 hops away, so the edge like `a -- f` will never be generated, because `f` is too far away.

Does that make things a bit clearer?

**Comment by loySoGxj** [comment]
Yes, it's clear.
There is a possibility that uniform sampling can be devised a little more.
It was very helpful for me to explain it carefully.
I would even like to emulate the speed of this response.
If you find you on twitter, I'll follow you.
I want to close it with this.
